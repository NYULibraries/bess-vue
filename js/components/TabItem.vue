<template>
  <li>
    <a :href="tab.open && tab.open.href || '#'"
      :title="tab.title"
      :target="tab.open && tab.open.target"
      @click="updateTab($event, tab)"
    ><slot></slot></a>
  </li>
</template>

<script>
  export default {
    name: 'tab-item',
    props: ['tab', 'updateTab'],
  };
</script>
