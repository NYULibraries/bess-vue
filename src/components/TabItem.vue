<template>
  <li>
    <a
      :href="tab.open && tab.open.href || '#'"
      :title="tab.title"
      :target="tab.open && tab.open.target"
      @click="updateTab( $event, tab )"
    ><slot /></a>
  </li>
</template>

<script>
export default {
    name : 'TabItem',
    props: [ 'tab', 'updateTab' ],
};
</script>
