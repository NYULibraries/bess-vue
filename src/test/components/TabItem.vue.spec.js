import { afterEach, beforeEach, describe, expect, test, vi } from 'vitest';
import { shallowMount } from '@vue/test-utils';

import appConfig from '../../../config';
import TabItem from '@/components/TabItem.vue';

const updateTabSpy = vi.fn( () => 0 );

describe( 'TabItem', () => {
    let currentProps;
    let wrapper;

    describe.each(
        Object.keys( appConfig.institutions ).map(
            function ( institution ) {
                return { institution };
            },
        ) )( '$institution', ( { institution } ) => {
        describe.each(
            appConfig.institutions[ institution ].map(
                function ( tabConfig ) {
                    return { tabConfig };
                },
            ) )( '$tabConfig', ( { tabConfig } ) => {
            beforeEach( () => {
                currentProps = {
                    tab: {
                        title: tabConfig.title,
                        open : tabConfig.open,
                    },
                    updateTab: updateTabSpy,
                };

                wrapper = shallowMount( TabItem, {
                    props: currentProps,
                } );
            } );

            test( 'is a Vue instance', () => {
                expect( wrapper ).toBeTruthy();
            } );

            test( 'has the correct HTML', () => {
                expect( wrapper.html() ).toMatchSnapshot();
            } );

            describe( 'props', () => {
                test( 'should receive tab and updateTab properties', () => {
                    expect( wrapper.props().tab ).toEqual( currentProps.tab );
                    expect( wrapper.props().updateTab ).toEqual( updateTabSpy );
                } );
            } );

            describe( 'click on tab link', () => {
                afterEach( () => {
                    vi.resetAllMocks();
                } );

                test( 'should trigger updateTab', () => {
                    wrapper.find( 'a' ).trigger( 'click' );
                    expect( updateTabSpy ).toHaveBeenCalled();
                } );

                test( 'should not trigger on click of outer element', () => {
                    wrapper.trigger( 'click' );
                    expect( updateTabSpy ).not.toHaveBeenCalled();
                } );
            } );
        } );
    } );
} );
