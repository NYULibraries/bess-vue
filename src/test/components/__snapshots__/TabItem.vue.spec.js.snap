// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TabItem > 'NYU' > { label: 'Catalog Search', …(4) } > has the correct HTML 1`] = `"<li><a href="#" title="Search NYU's catalog for books, journals, scripts, scores, archival materials, NYU dissertations, videos, sound recordings"></a></li>"`;

exports[`TabItem > 'NYU' > { label: 'Course Reserves', …(2) } > has the correct HTML 1`] = `"<li><a href="https://ares.library.nyu.edu/" title="Search for library materials that are held at one location for a particular course" target="_blank"></a></li>"`;

exports[`TabItem > 'NYU' > { label: 'Databases', …(2) } > has the correct HTML 1`] = `"<li><a href="http://guides.nyu.edu/arch" title="Search databases for articles or browse databases by subject" target="_blank"></a></li>"`;

exports[`TabItem > 'NYU' > { label: 'Research Guides', …(2) } > has the correct HTML 1`] = `"<li><a href="https://guides.nyu.edu" title="Guides to help you find library resources on specific subjects and courses" target="_blank"></a></li>"`;

exports[`TabItem > 'NYU_HOME' > { label: 'Articles & Databases', …(2) } > has the correct HTML 1`] = `"<li><a href="http://guides.nyu.edu/arch" title="Search databases for articles or browse databases by subject" target="_blank"></a></li>"`;

exports[`TabItem > 'NYU_HOME' > { label: 'Catalog Search', …(4) } > has the correct HTML 1`] = `"<li><a href="https://search.library.nyu.edu/discovery/search?vid=01NYU_INST:NYU_DEV" title="Search NYU's catalog for books, journals, scripts, scores, archival materials, NYU dissertations, videos, sound recordings" target="_blank"></a></li>"`;

exports[`TabItem > 'NYU_HOME' > { label: 'Course Reserves', …(2) } > has the correct HTML 1`] = `"<li><a href="https://ares.library.nyu.edu/" title="Search for library materials that are held at one location for a particular course" target="_blank"></a></li>"`;

exports[`TabItem > 'NYUAD' > { label: 'Catalog Search', …(4) } > has the correct HTML 1`] = `"<li><a href="#" title="Search NYU's catalog for books, journals, scripts, scores, archival materials, NYU dissertations, videos, sound recordings"></a></li>"`;

exports[`TabItem > 'NYUAD' > { label: 'Course Reserves', …(3) } > has the correct HTML 1`] = `"<li><a href="#" title="Search for library materials that are held at one location for a particular course"></a></li>"`;

exports[`TabItem > 'NYUAD' > { label: 'Databases', …(2) } > has the correct HTML 1`] = `"<li><a href="http://guides.nyu.edu/adarch" title="Search databases for articles or browse databases by subject" target="_blank"></a></li>"`;

exports[`TabItem > 'NYUAD' > { label: 'My Accounts', …(2) } > has the correct HTML 1`] = `"<li><a href="#" title="My Accounts"></a></li>"`;

exports[`TabItem > 'NYUAD' > { label: 'Research Guides', …(3) } > has the correct HTML 1`] = `"<li><a href="#" title="Guides to help you find library resources on specific subjects and courses"></a></li>"`;

exports[`TabItem > 'NYUSH' > { label: 'Catalog Search', …(4) } > has the correct HTML 1`] = `"<li><a href="#" title="Search NYU's catalog for books, journals, scripts, scores, archival materials, NYU dissertations, videos, sound recordings"></a></li>"`;

exports[`TabItem > 'NYUSH' > { label: 'Course Reserves', …(2) } > has the correct HTML 1`] = `"<li><a href="https://ares.library.nyu.edu/" title="Search for library materials that are held at one location for a particular course" target="_blank"></a></li>"`;

exports[`TabItem > 'NYUSH' > { label: 'Databases', …(2) } > has the correct HTML 1`] = `"<li><a href="http://guides.nyu.edu/arch" title="Search databases for articles or browse databases by subject" target="_blank"></a></li>"`;

exports[`TabItem > 'NYUSH' > { label: 'My Accounts', …(2) } > has the correct HTML 1`] = `"<li><a href="#" title="My Accounts"></a></li>"`;

exports[`TabItem > 'NYUSH' > { label: 'Research Guides', …(3) } > has the correct HTML 1`] = `"<li><a href="#" title="Guides to help you find library resources on specific subjects and courses"></a></li>"`;
